business/health.go

```diff
@@ -114,7 +117,7 @@ func (in *HealthService) GetWorkloadHealth(ctx context.Context, namespace, workl
 }
 
 // GetNamespaceAppHealth returns a health for all apps in given Namespace (thus, it fetches data from K8S and Prometheus)
-func (in *HealthService) GetNamespaceAppHealth(ctx context.Context, criteria NamespaceHealthCriteria) (models.NamespaceAppHealth, error) {
+func (in *HealthService) GetNamespaceAppHealth(ctx context.Context, criteria NamespaceHealthCriteria) (models.NamespaceAppsHealth, error) {
```

Comment:
I know we're not doing this on other pages because those are driven off the aggregated namespace selector but for the overview page and the namespace health, can you just send cluster as a query param through `NamespaceHealthCriteria`. Then, I think, we can keep all the models the same `models.NamespaceAppHealth` vs. `models.NamespaceAppsHealth` and we wouldn't need to loop over all the clusters. For single cluster, since cluster is a query param, if it's not provided or `""` then we can assume you want the home cluster.

Reply:
I thought of that before, it can be done I guess. 

When I decided to go this path, was to be consistent with the same logic we are using in the list pages (this is not a list page but it's similar, you have namespaces and health for n applications, so we are listing health among other things, just that the UI show them in a different aggregated way). There is no namespace selector because we are usually fetching all the aggregated namespaces.

I need to check if there is always the cluster information available in the UI, I guess so, because I added the field to the Namespace model, but I need to check in terms of the state, if I won't have a problem there. 

Having said this, maybe we need to do this (cluster as query param) in the lists too?.

Reply:
> Having said this, maybe we need to do this (cluster as query param) in the lists too?.

Yes we should but it's more of a pain there. Those pages make api requests per namespace based off of the namespace selector which currently aggregates all the namespaces across all the clusters. Since they're aggregated you lose the cluster info that goes along with the namespace object. In addition, those pages sometimes save their state in the URL where namespace is only saved as its name. We should update those to pass cluster as a query param but it's a bigger lift right now.

With the overview page, I don't think you have those problems. It just gets the namespaces from the `/api/namespaces` api call instead of the namespace selector which should return a `Namespace` object with the cluster since `/api/namespaces` returns the namespaces for every cluster. Also I don't think there's state saved in the URL for the namespaces page. It would be good to be consistent across pages but imo we should consistently pass cluster as a query param rather than trying to talk to every cluster on the backend for each API call.

Reply:
I also checked in the appenders, because the namespace health is used there too, and we should have the cluster name because the Node object has it according to:

https://github.com/kiali/kiali/commit/6ec37a53ddbcc88ee55959aeeac7114b33f893aa

So, I will refactor this PR to use the cluster as the query param, no problem.

Reply:
Hi @nrfox, I refactored the code to use the cluter as query parameter.